File: RADWIMPS.ts
class RADWIMPS {
  constructor(private buf: string[] = []) {}

  then() {
    return new RADWIMPS([...this.buf, '前']);
  }

  世() {
    console.log([...this.buf, '世'].join(''));
  }
}

const radwimps = new RADWIMPS();
radwimps.then().then().then().世();

File: RADWIMPS.hs
main :: IO ()
main = radwimps $
  then'().then'().then'().世()

radwimps :: (() -> IO ()) -> IO ()
radwimps = ($ ())

then' :: () -> IO a -> IO a
then' () k = putStr "前" >> k

世 :: () -> () -> IO ()
世 () () = putStrLn "世"
